### PR TITLE
doc: Clarifies private endpoint resource docs

### PR DESCRIPTION
**Pull request by zach-carr**
## Description

The resources mongodbatlas_privatelink_endpoint_service and mongodbatlas_privatelink_endpoint map to contradictory terminology -- privatelink_endpoint_service affects Private Endpoints, and privatelink_endpoint affects Private Endpoint Services. To help clarify, I've added links to a new section of the Atlas documentation that defines these terms as MongoDB uses them.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: website/docs/r/privatelink_endpoint_service.html.markdown [hunk 1]
```diff
@@ -8,13 +8,14 @@ description: |-
...
 
-`mongodbatlas_privatelink_endpoint_service` provides a Private Endpoint Interface Link resource. This represents a Private Endpoint Interface Link, which adds one interface endpoint to a private endpoint connection in an Atlas project.
+`mongodbatlas_privatelink_endpoint_service` provides a Private Endpoint Interface Link resource. This represents a Private Endpoint Interface Link, which adds one [Interface Endpoint](https://www.mongodb.com/docs/atlas/security-private-endpoint/#private-endpoint-concepts) to a private endpoint connection in an Atlas project.
 
 ~> **IMPORTANT:**You must have one of the following roles to successfully handle the resource:
   * Organization Owner
   * Project Owner
 
 -> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.
+
```
Comment (zach-carr) [inline website/docs/r/privatelink_endpoint_service.html.markdown:18]: unrelated fix to note rendering

**Review by sarahsimpers — APPROVED** [review]
LGTM, thank you!

**Review by oarbusi — APPROVED** [review]
Thanks! LGTM
